### DISCUSSION TITLE
History activity log

### DISCUSSION THREAD

**Discussion by twobit** [liveblocks/liveblocks#1365] [Help] [answered]
Is there a way to get the history of a room to show an activity log? Basically, who made what changes and when?

**Comment by CTNicholas** [comment, marked as answer]
Hi @twobit, we don't currently have an activity log. When it comes to Storage and Presence, each user's history is stored locally in their browser, so an activity log would require a lot of changes. 

What would you to build with it? We'd love to learn about your use case 🙂

**Reply by twobit** [reply]
I'd like to allow users to restore a room to an earlier state. It would need to be possible to restore beyond a browser session. Maybe if Storage supported something like checkpoints.

**Reply by therealtimhawkins** [reply]
I'd also be really interested in a feature like this. I'd like to be able to show the history of the room as a list of changes made by each user. Sort of like a commit history in git. Just thought I'd drop this in! 😅

**Reply by shellscape** [reply]
+1 - we're resorting to leveraging the webhook to produce this ourselves. would much prefer treating liveblocks as the defacto storage for our room data.

**Comment by metav-drimz** [comment]
any update on the matter? plus where exactly the local history data is stored?
